demos/tilemap/tilemap.go

```diff
@@ -1,54 +1,96 @@
 }
 
 func (game *GameWorld) Preload() {
-	game.New()
-	engo.Files.Add("data/sheet.png")
+	// A tmx file can be generated from the Tiled Map Editor.
+	// The engo tmx loader only accepts tmx files that are base64 encoded and compressed with zlib.
+	// When you add tilesets to the Tiled Editor, the location where you added them from is where the engo loader will look for them
+	// Tileset from : http://opengameart.org
 
-	log.Println("Preloaded")
+	if err := engo.Files.LoadMany("example.tmx"); err != nil {
+		panic(err)
+	}
 }
 
-func (game *GameWorld) Setup() {
-	common.SetBackground(0x2d3739)
+func (game *GameWorld) Setup(w *ecs.World) {
+	common.SetBackground(color.RGBA{0x00, 0x00, 0x00, 0x00})
 
 	w.AddSystem(&common.RenderSystem{})
 
-	gameMap := ecs.NewEntity("RenderSystem")
-	tilemap := engo.NewTilemap(
-		[][]string{
-			{"0", "2", "0"},
-			{"4", "5", "1"},
-			{"2", "3", "4"},
-			{"5", "1", "2"}},
-		engo.Files.Image("sheet"), 16)
-
-	mapRender := engo.NewRenderComponent(tilemap, engo.Point{1, 1}, "map")
-	mapSpace := &common.SpaceComponent{engo.Point{100, 100}, 0, 0}
-	gameMap.AddComponent(mapRender)
-	gameMap.AddComponent(mapSpace)
-
-	err := w.AddEntity(gameMap)
+	resource, err := engo.Files.Resource("example.tmx")
 	if err != nil {
-		log.Println(err)
+		panic(err)
+	}
+	tmxResource := resource.(common.TMXResource)
+	levelData := tmxResource.Level
+
+	// Create render and space components for each of the tiles
+	tileComponents := make([]*Tile, 0)
+	for _, v := range levelData.Tiles {
+		if v.Image != nil {
+			tile := &Tile{BasicEntity: ecs.NewBasic()}
+			tile.RenderComponent = common.RenderComponent{
+				Drawable: v,
+				Scale:    engo.Point{1, 1},
+			}
+			tile.SpaceComponent = common.SpaceComponent{
+				Position: v.Point,
+				Width:    0,
+				Height:   0,
+			}
+			tileComponents = append(tileComponents, tile)
+		}
+	}
+	// Do the same the levels images
+	for _, v := range levelData.Images {
+		if v.Image != nil {
+			tile := &Tile{BasicEntity: ecs.NewBasic()}
+			tile.RenderComponent = common.RenderComponent{
+				Drawable: v,
+				Scale:    engo.Point{1, 1},
+			}
+			tile.SpaceComponent = common.SpaceComponent{
+				Position: v.Point,
+				Width:    0,
+				Height:   0,
+			}
+			tileComponents = append(tileComponents, tile)
+		}
 	}
+
+	// Add each of the tiles entities and its components to the render system
+	for _, system := range w.Systems() {
+		switch sys := system.(type) {
+		case *common.RenderSystem:
+			for _, v := range tileComponents {
+				sys.Add(&v.BasicEntity, &v.RenderComponent, &v.SpaceComponent)
+			}
+
+		}
+	}
+
 }
 
+func (game *GameWorld) Exit()        {}
+func (game *GameWorld) Hide()        {}
+func (game *GameWorld) Show()        {}
+func (game *GameWorld) Type() string { return "GameWorld" }
+
 func main() {
-	World = &GameWorld{}
-	engo.Run("Hello", 1024, 640, false, World)
+	engo.Run(engo.RunOptions{Title: "TileMap Demo", Width: 800, Height: 800, ScaleOnResize: false}, &GameWorld{})
```

Comment:
One last thing is to try to get this bit to match with the other demos. For example: https://github.com/EngoEngine/engo/blob/master/demos/hello-world/hello.go#L62